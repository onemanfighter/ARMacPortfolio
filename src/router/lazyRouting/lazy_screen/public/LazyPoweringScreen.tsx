import { lazy } from 'react';
import { LazyProvider } from '../../component';

const Powering = lazy(
  () => import('../../../../screens/public/Powering/Powering'),
);

const LazyPoweringComponent = () => {
  return (
    <LazyProvider>
      <Powering />
    </LazyProvider>
  );
};

export { LazyPoweringComponent };
