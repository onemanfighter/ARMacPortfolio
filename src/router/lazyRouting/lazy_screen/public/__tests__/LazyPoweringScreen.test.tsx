import { render, screen, waitFor } from '@testing-library/react';
import { LazyPoweringComponent } from '../LazyPoweringScreen';
import { uiStore } from '@uiStore';
import { renderHook } from '@testing-library/react-hooks';

describe('LazyPoweringComponent', () => {
  it('should render correctly to match snapshot', async () => {
    const { result } = renderHook(() => uiStore());
    const { container } = render(<LazyPoweringComponent />);

    await waitFor(() =>
      expect(screen.getByLabelText('powering-screen')).toBeDefined(),
    );
    expect(container).toMatchSnapshot();
  });
});
