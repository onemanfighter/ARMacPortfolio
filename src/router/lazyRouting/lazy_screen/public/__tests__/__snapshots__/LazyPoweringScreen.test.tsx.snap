// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyPoweringComponent should render correctly to match snapshot 1`] = `
<div>
  <div>
    <h1>
      Powering
    </h1>
  </div>
</div>
`;
